File: Intermediate/06_power.c
#include <stdio.h>
#include <math.h>
int isPowerof(int x, int y);//fucntion prototpe
//main
int main()
{
    int x, y;
    printf("Enter the number ");
    scanf("%d", &x);
    printf("Enter the power to check");
    scanf("%d", &y);
    (isPowerOf(x, y) == 1) ? printf("True") : printf("False");
    return 0;
}
//def of fuction
int isPowerOf(int x, int y)
{
    //base change formula (property of log)
    /*logab*/
    double p = log10(x) / log10(y);
    if (p - (int)p == 0)
    {
        return 1;
    }
    else
    {
        return 0;
    }
    return 0;
}

File: Intermediate/06_power.py
import math
class Solution:
    def isPower(self,n:int,x:int) ->bool:
        # if n <=0:
        #     return False
        p=math.log10(n)/math.log10(x)
        if (p-int(p))==0:
            return True
        else:
            return False
    
x=int(input(("Enter the number ")))
y=int(input(("Enter the power to check")))
print(Solution().isPower(x,y))